lib/puppet/provider/mysql_datadir/mysql.rb

```diff
@@ -34,15 +35,15 @@ def create
     end
 
     if mysqld_version.nil?
-      debug("Installing MySQL data directory with mysql_install_db --basedir=#{basedir} #{defaults_extra_file} --datadir=#{datadir} --user=#{user}")
-      mysql_install_db(["--basedir=#{basedir}",defaults_extra_file, "--datadir=#{datadir}", "--user=#{user}"].compact)
+      debug("Installing MySQL data directory with mysql_install_db #{defaults_extra_file} --basedir=#{basedir} --datadir=#{datadir} --user=#{user}")
+      mysql_install_db([defaults_extra_file, "--basedir=#{basedir}", "--datadir=#{datadir}", "--user=#{user}"].compact)
     else
-      if mysqld_type == "mysql" and Puppet::Util::Package.versioncmp(mysqld_version, '5.7.6') >= 0
+      if (mysqld_type == "mysql" or mysqld_type == "percona") and Puppet::Util::Package.versioncmp(mysqld_version, '5.7.6') >= 0
```

Comment:
Hmm. We change this in 3 separate places in 2 files so I'm thinking we should really define a predicate function with this condition in it, otherwise I worry that other people will continue to ignore percona when making changes. Going to look around and see where might be a good idea to define that function.